pkg/vault/secretmanager.go

```diff
@@ -28,7 +43,71 @@ func (s *SecretManager) GetSecrets(path string) (map[string]interface{}, error)
 		return nil, err
 	}
 
-	// Will need to do more to handle the data
+	// Make sure the secret exists
+	if _, ok := data["secrets"]; !ok {
+		return nil, fmt.Errorf("Could not find secrets at path %s", path)
+	}
+
+	// Get list of secrets
+	secretList := data["secrets"].([]interface{})
+	v := make([]string, 0, len(secretList))
+	// Loop through secrets and get id
+	// as getting the list of secrets does not include the payload
+	for _, value := range secretList {
+		secret := value.(map[string]interface{})
+		if t, found := secret["id"]; found {
+			v = append(v, t.(string))
+		}
+	}
+
+	// Read each secret and get payload
+	secrets := make(map[string]interface{})
+	for _, j := range v {
+		data, err := s.Client.Read(fmt.Sprintf("%s/%s", path, j))
+		if err != nil {
+			return nil, err
+		}
+		// Get name and data of secret and append to secrets map
+		secretName := data["name"].(string)
+		secretData := data["secret_data"].(map[string]interface{})
+		secrets[secretName] = secretData["payload"]
+	}
+
+	return secrets, nil
+}
+
+func getAccessToken(apikey string) (string, error) {
+	// Set url values to be added to the request
+	urlValues := url.Values{}
+	urlValues.Set("grant_type", "urn:ibm:params:oauth:grant-type:apikey")
+	urlValues.Add("apikey", apikey)
+
+	// Creating request to get access token
+	req, err := http.NewRequest("POST", "https://iam.cloud.ibm.com/identity/token", strings.NewReader(urlValues.Encode()))
```

Comment:
Should these strings be constants? Not a big deal

Reply:
Made an issue for this